src/protocols/secure_channel/CASESession.cpp

```diff
@@ -514,7 +514,8 @@ CHIP_ERROR CASESession::EstablishSession(SessionManager & sessionManager, Fabric
 
     // We are setting the exchange context specifically before checking for error.
     // This is to make sure the exchange will get closed if Init() returned an error.
-    mExchangeCtxt = exchangeCtxt;
+    ExchangeHandle ecHandle(*exchangeCtxt);
+    mExchangeCtxt.SetValue(ecHandle);
```

Comment:
```suggestion
    mExchangeCtxt.SetValue(ExchangeHandle(*exchangeCtxt));
```

Reply:
`mExchangeCtxt.Emplace(*exchangeCtxt)` could also work and avoid calling the copy constructor.